### PR TITLE
WIP: CocoaPods readme

### PR DESCRIPTION
**Pull request by ilmat192**

### CODE REVIEW

File: COCOAPODS.md [hunk 1]
```diff
@@ -0,0 +1,148 @@
+## CocoaPods integration
+
+### Introduction
+
+Starting with 1.3.30 an experimental integration with [CocoaPods](https://cocoapods.org/) is added
+in Kotlin/Native. This feature allows one to represent a Kotlin/Native Gradle-project as a
```
Comment (olonho) [inline COCOAPODS.md]: added to

File: COCOAPODS.md [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+CocoaPods dependency. Such a representation provides the following advantages:
+
+ - Such a dependency can be included in a Podfile of an Xcode project and be automatically (re)built
+ along with this project. As a result, importing to Xcode is simplified since there is no need to
+ write corresponding Gradle tasks and Xcode build steps manually.
+ 
+ - When building from Xcode, it becomes possible to use CocoaPods libraries without manual writing
```
Comment (olonho) [inline COCOAPODS.md]: becomes -> is

File: COCOAPODS.md [hunk 3]
```diff
@@ -0,0 +1,148 @@
...
+ def-files and setting cinterop tool parameters. In this case all required parameters can be
```
Comment (olonho) [inline COCOAPODS.md]: .def files

File: COCOAPODS.md [hunk 4]
```diff
@@ -0,0 +1,148 @@
...
+ 
+### CocoaPods Gradle plugin
+
+The CocoaPods support in implemented in a separate Gradle plugin: `org.jetbrains.kotlin.native.cocoapods`.
+
+> __Note:__ The plugin is based on the multiplatform project model and requires applying the
+`org.jetbrains.kotlin.multiplatform` plugin. See details about the multiplatform plugin at
+the [corresponding page](https://kotlinlang.org/docs/reference/building-mpp-with-gradle.html).
+
+When applied the CocoaPods plugin preforms the following actions:
```
Comment (olonho) [inline COCOAPODS.md]: performs

File: COCOAPODS.md [hunk 5]
```diff
@@ -0,0 +1,148 @@
...
+
+#####TODO: Sample link (https://github.com/JetBrains/kotlin-native/pull/2827).
+
+<div class="sample" markdown="1" theme="idea" mode="kotlin">
+
+```kotlin
+// Apply plugins.
+plugins {
+    id("org.jetbrains.kotlin.multiplatform")
+    id("org.jetbrains.kotlin.native.cocoapods")
```
Comment (jonnyzzz) [inline COCOAPODS.md]: version is missing (both for multiplatform and cocoapods).

**Comment by jonnyzzz** [comment]
You should list the new document in the https://github.com/JetBrains/kotlin-native/blob/master/_nav_reference.yml file

File: COCOAPODS.md [hunk 6]
```diff
@@ -0,0 +1,150 @@
...
+add dependencies on CocoaPods libraries.
+
+<div class="sample" markdown="1" theme="idea" mode="kotlin">
+
+```kotlin
+kotlin {
+    cocoapods {
+        // Configure a dependency on AFNetworking.
+        // The CocoaPods version notation is supported.
+        // The dependency will be added to all macOS and iOS targets.
```
Comment (olonho) [inline COCOAPODS.md]: Can we control which target  dependency is added to?

Comment (ilmat192) [reply COCOAPODS.md]: Currently no.

File: COCOAPODS.md [hunk 7]
```diff
@@ -0,0 +1,150 @@
...
+> A version of the Gradle project is used as a value for the `version` field.
+Fields`summary` and `homepage` can be configured using the `cocoapods` code block.
+
+This podspec file can be referenced from a [Podfile](https://guides.cocoapods.org/using/the-podfile.html)
+of an Xcode project. After that the framework built from the Kotlin/Native module can be used from
+this Xcode project. If necessary, this framework is automatically rebuilt during Xcode build process.
+
+### Workflow
+
+The following steps allows one to import a Kotlin/Native module in an existing Xcode project.
```
Comment (p7nov) [inline COCOAPODS.md]:
```suggestion
To import a Kotlin/Native module in an existing Xcode project:
```

File: COCOAPODS.md [hunk 8]
```diff
@@ -0,0 +1,150 @@
...
+
+0. [Install](https://guides.cocoapods.org/using/getting-started.html#installation) CocoaPods.
```
Comment (p7nov) [inline COCOAPODS.md]: Make sure that you have CocoaPods [installed](https://guides.cocoapods.org/using/getting-started.html#installation).

File: COCOAPODS.md [hunk 9]
```diff
@@ -0,0 +1,150 @@
...
+It's recommended to use CocoaPods 1.6.1 or higher.
```
Comment (p7nov) [inline COCOAPODS.md]:
```suggestion
We recommend using CocoaPods 1.6.1 or later.
```

**Pull request merged**